File: PE002-Even Fibonacci numbers/PE002.cpp
#include <iostream>

int main() {
  unsigned int first = 1;
  unsigned int second = 1;
  unsigned int evens_sum = 0;

  while (second <= 4000000) {
    unsigned int sum = first + second;
    if (sum % 2 == 0) {
      evens_sum += sum;
    }

    second = first;
    first = sum;
  }

  std::cout << evens_sum << "\n";

  return 0;
}

File: PE002-Even Fibonacci numbers/PE002.rs
fn main() {
  let mut first = 1;
  let mut second = 1;
  let mut evens_sum = 0;

  while second <= 4_000_000 {
    let sum = first + second;
    if sum % 2 ==  0 {
      evens_sum += sum;
    }

    second = first;
    first = sum;
  }

  println!("{}", evens_sum);
}
